src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingAssessmentIntegrationTest.java

```diff
@@ -676,6 +678,48 @@ void cancelAssessmentOfOtherTutorAsInstructor() throws Exception {
         cancelAssessment(HttpStatus.OK);
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
+    void testSaveAssessmentNote() throws Exception {
+        AssessmentNote assessmentNote = new AssessmentNote();
+        assertThat(assessmentNote.getCreatedDate()).isNotNull();
+
+        assessmentNote.setNote("note");
+        manualResult.setAssessmentNote(assessmentNote);
+
+        User user = userUtilService.getUserByLogin(TEST_PREFIX + "tutor1");
+        manualResult.setAssessor(user);
+        assertThat(manualResult.getAssessmentNote().getCreator()).isNull();
+
+        manualResult = request.putWithResponseBody("/api/participations/" + manualResult.getParticipation().getId() + "/manual-results", manualResult, Result.class, HttpStatus.OK);
+        assertThat(manualResult.getAssessmentNote().getCreator()).isNotNull().isEqualTo(user);
+
+        Instant createdDate = assessmentNote.getCreatedDate();
+        assertThat(createdDate).isNotNull();
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
+    void testAssessmentNoteTimestamps() throws Exception {
+        AssessmentNote assessmentNote = new AssessmentNote();
+        assessmentNote.setNote("note1");
+        manualResult.setAssessmentNote(assessmentNote);
+        manualResult = request.putWithResponseBody("/api/participations/" + manualResult.getParticipation().getId() + "/manual-results", manualResult, Result.class, HttpStatus.OK);
+
+        Instant createdDate = manualResult.getAssessmentNote().getCreatedDate();
+        Instant initialLastModifiedDate = manualResult.getAssessmentNote().getLastModifiedDate();
+
+        // Because the two timestamps are created sequentially, it is required to compare them based on some epsilon,
+        // which is a second here. Going too low might make the test flaky.
+        Duration difference = Duration.between(createdDate, initialLastModifiedDate);
+        assertThat(difference.abs().toSeconds()).isNotPositive();
```

Comment:
I think using !Instant.isBefore analogous to line 720 is more elegant. The epsilon can be implemented using the Instant.plus method.

Reply:
Completely improved the test since there were some other issues as well with the auditing annotations in the assessment note.